aws-predictions/src/main/java/com/amplifyframework/predictions/aws/service/AWSComprehendService.java

```diff
@@ -0,0 +1,338 @@
+import com.amplifyframework.predictions.result.InterpretResult;
+import com.amplifyframework.util.UserAgent;
+
+import com.amazonaws.AmazonClientException;
+import com.amazonaws.ClientConfiguration;
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.mobile.client.AWSMobileClient;
+import com.amazonaws.services.comprehend.AmazonComprehendClient;
+import com.amazonaws.services.comprehend.model.DetectDominantLanguageRequest;
+import com.amazonaws.services.comprehend.model.DetectDominantLanguageResult;
+import com.amazonaws.services.comprehend.model.DetectEntitiesRequest;
+import com.amazonaws.services.comprehend.model.DetectEntitiesResult;
+import com.amazonaws.services.comprehend.model.DetectKeyPhrasesRequest;
+import com.amazonaws.services.comprehend.model.DetectKeyPhrasesResult;
+import com.amazonaws.services.comprehend.model.DetectSentimentRequest;
+import com.amazonaws.services.comprehend.model.DetectSentimentResult;
+import com.amazonaws.services.comprehend.model.DetectSyntaxRequest;
+import com.amazonaws.services.comprehend.model.DetectSyntaxResult;
+import com.amazonaws.services.comprehend.model.DominantLanguage;
+import com.amazonaws.services.comprehend.model.PartOfSpeechTag;
+import com.amazonaws.services.comprehend.model.SentimentScore;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Predictions service for performing text interpretation.
+ */
+final class AWSComprehendService {
+
+    private final AmazonComprehendClient comprehend;
+    private final AWSInterpretConfiguration configuration;
+
+    AWSComprehendService(@Nullable AWSInterpretConfiguration configuration) throws PredictionsException {
+        this.comprehend = createComprehendClient();
+        this.configuration = configuration;
+    }
+
+    private AmazonComprehendClient createComprehendClient() {
+        AWSCredentialsProvider credentialsProvider = AWSMobileClient.getInstance();
+        ClientConfiguration configuration = new ClientConfiguration();
+        configuration.setUserAgent(UserAgent.string());
+        return new AmazonComprehendClient(credentialsProvider, configuration);
+    }
+
+    void comprehend(
+            @NonNull String text,
+            @NonNull Consumer<InterpretResult> onSuccess,
+            @NonNull Consumer<PredictionsException> onError
+    ) {
+        try {
+            // First obtain the dominant language to begin analysis
+            final Language dominantLanguage = fetchPredominantLanguage(text);
+            final LanguageType language = dominantLanguage.getValue();
+
+            // Actually analyze text in the context of dominant language
+            final Sentiment sentiment = fetchSentiment(text, language);
+            final List<KeyPhrase> keyPhrases = fetchKeyPhrases(text, language);
+            final List<Entity> entities = fetchEntities(text, language);
+            final List<Syntax> syntax = fetchSyntax(text, language);
```

Comment:
That... is actually exactly what I should have been going for haha

Reply:
on second thought, I think I'll revisit this in the future. This may require each component to be made into its own predictions service method, and I want to see how the final Predictions service with all of the APIs implemented looks before I do that.